distributed/protocol/tests/test_numpy.py

```diff
@@ -82,18 +79,20 @@ def test_dumps_serialize_numpy(x):
         assert x.strides == y.strides
 
 
-def test_masked_array_serialize():
-    data = (5, 6)
-    mask = [True, False]
-    fill_value = 999
-    x = np.ma.masked_array(data, mask=mask, fill_value=fill_value)
-    header, frames = serialize(x)
-    y = deserialize(header, frames)
-
-    # Explicitly test the particular elements of the masked array.
-    np.testing.assert_equal(data, y.data)
-    np.testing.assert_equal(mask, y.mask)
-    assert fill_value == y.fill_value
+@pytest.mark.parametrize('x', [
+    np.ma.masked_array([5, 6], mask=[True, False], fill_value=10, dtype='i4'),
+    np.ma.masked_array([5., 6.], mask=[True, False], fill_value=10, dtype='f4'),
+    np.ma.masked_array([5., 6.], mask=[True, False], fill_value=np.nan, dtype='f8'),
+    np.ma.masked_array([5., 6.], mask=np.ma.nomask, fill_value=np.nan, dtype='f8'),
+    np.ma.masked_array([True, False], mask=np.ma.nomask, fill_value=True, dtype='bool'),
+    np.ma.masked_array(['a', 'b'], mask=[True, False], fill_value='c', dtype='O')
```

Comment:
One more parameter that would be interesting to see here is 
```python
np.ma.masked_array([5, 6], mask=True, fill_value=10)
```
We've occasionally seen some odd behaviour stemming from NumPy in this particular case, which is something to do with how NumPy interprets the mask. I've just tried creating one of these locally, however, and it doesn't seem to be that unusual, so perhaps this has been improved in more recent NumPy versions.

You could also consider checking the default fill-values for different dtypes, which was implicitly being done before by making use of the default fill-values in the first two masked array parameters. Given, however, that these checks are effectively being done by the first two parameters here, and I doubt there's much difference between serializing `10.0` and `1.0e+20`, this probably isn't too important.

Reply:
Apologies, I didn't demonstrate the situation I was thinking of correctly. Here's a functioning example, produced by not specifying the mask at all when creating a masked array:

```python
>>> np.ma.arange(10)
masked_array(data = [0 1 2 3 4 5 6 7 8 9],
             mask = False,
       fill_value = 999999)
```

(For reference, the same behaviour is also present using `dask.array.ma`.)

Reply:
@dkillick just to make sure I understand, is there a test that you think should be added here?

From what I read above it sounds like you didn't actually find a concrete issue.  I may be misreading though and want to check in before I hit merge here.

Reply:
@mrocklin sorry I missed your comment. There's no concrete issue that I know of here, just the chance of some unexpected NumPy behaviour that I thought it might be good to test.